### DISCUSSION TITLE
RAUC on intel x86_64 systemd-boot uefi

### DISCUSSION THREAD

**Discussion by ghuband** [rauc/rauc#1168] [Q&A]
I'm trying to setup RAUC on a x86_64 system which uses systemd-boot and uefi.  I'm using Yocto kirkstone and the meta-intel layer.  I've followed the RAUC instructions for EFI setup (https://rauc.readthedocs.io/en/latest/integration.html#efi).  It looks like systemd-boot (https://www.man7.org/linux/man-pages/man7/systemd-boot.7.html, under BOOT COUNTING) can track the boot attempts and fallback to the last good boot.  Does RAUC use this feature of systemd-boot?  Do I need the systemd-bless-boot.service running?

**Comment by ejoerns** [comment]
@ghuband The approach RAUC supports works by directly setting EFI boot variables via `efibootmgr`. Explicit support for `systemd-boot` is still missing and I am not that up-to-date on the topic that I'd know how compatible this is. But I can at least point you to #565 where the latest discussion about this happened.

Anyway, supporting systemd-boot I still see as a relevant topic.

**Reply by ghuband** [reply]
Thanks.  What EFI boot variables does RAUC set?

**Reply by ejoerns** [reply]
It sets the standard UEFI Boot Manager variables. It does nothing more than calling `efibootmgr`. Thus you might want to have a look at this to see which variables these are exactly. I guess the corresponding UEFI doc is https://uefi.org/specs/UEFI/2.10/03_Boot_Manager.html.

For some details about what RAUC invokes, maybe https://rauc.readthedocs.io/en/latest/reference.html#efi might help.

**Reply by ghuband** [reply]
OK, I think I see how RAUC works with the efi boot variables.  Does RAUC change the /boot/loader/entries/boot.conf file?

**Reply by jluebbe** [reply]
No, as `/boot/loader/entries/boot.conf` is not defined by EFI itself, the `efi` bootloader backend only uses efi boot variables.

**Reply by ghuband** [reply]
So, for my systemd-boot/efi setup RAUC supports the efi part but not the systemd-boot part.  I would need to provide a way to change the files needed by systemd-boot, which I think is changing the /boot/loader/entries/boot.conf file each time I do a "rauc install".

**Reply by jluebbe** [reply]
When using systemd-boot, RAUC should probably communicate with systemd-boot and *not* EFI, as the EFI-level boot entry will always point to systemd-boot, right?

So, it should probably be a different bootloader backend in RAUC.

How do you manage your kernel files in the ESP atomically? When using RAUC's EFI integration, usually each A/B slot has a corresponding partition with the kernel, so there is no chance of corrupting a shared partition during updates. Especially when using FAT (which has no journal), protection against power cuts is not guaranteed.
